### PR TITLE
[#134997] hide links unless permissions

### PR DESCRIPTION
**Pull request by meara**
- https://pm.tablexi.com/issues/134997

### CODE REVIEW

File: app/views/shared/_order_detail_cell.html.haml [hunk 1]
```diff
@@ -10,11 +10,12 @@
     - if od.product.offline?
       = tooltip_icon "icon-warning-sign", t("instruments.offline.note")
 
-  - if od.survey_completed?
-    .order-detail-extra= link_to 'View Order Form', od.external_service_receiver.show_url, :target => '_blank'
-  - od.stored_files.template_result.each do |stored_file|
-    .order-detail-extra
-      = link_to t("order_details.show.link.view.order_file"), facility_template_result_path(stored_file)
+  - if can?(:template_results, od)
```
Comment (meara) [inline app/views/shared/_order_detail_cell.html.haml]: It's a little tricky to find a permission that works for any external_service_receiver, but I figure if they shouldn't be able to see `template_results`, they probably shouldn't be able to see order forms either. Let me know if there is a better solution.

Comment (jhanggi) [inline app/views/shared/_order_detail_cell.html.haml]: I think that makes sense. I'm fine with it.

File: app/views/shared/_order_detail_cell.html.haml [hunk 2]
```diff
@@ -10,11 +10,12 @@
...
+    - if od.survey_completed?
+      .order-detail-extra= link_to 'View Order Form', od.external_service_receiver.show_url, :target => '_blank'
```
Comment (jhanggi) [inline app/views/shared/_order_detail_cell.html.haml]: While we're in here, can we lint this file?

**Comment by meara** [comment]
@jhanggi updated.

File: app/views/shared/_order_detail_cell.html.haml [hunk 3]
```diff
@@ -10,11 +10,12 @@
...
       = tooltip_icon "icon-warning-sign", t("instruments.offline.note")
 
-  - if od.survey_completed?
-    .order-detail-extra= link_to 'View Order Form', od.external_service_receiver.show_url, :target => '_blank'
-  - od.stored_files.template_result.each do |stored_file|
-    .order-detail-extra
-      = link_to t("order_details.show.link.view.order_file"), facility_template_result_path(stored_file)
+  - if can?(:template_results, od)
+    - if od.survey_completed?
+      .order-detail-extra= link_to 'View Order Form', od.external_service_receiver.show_url, target: '_blank'
```
Comment (jhanggi) [inline app/views/shared/_order_detail_cell.html.haml]: Looks like lint didn't pick up the single quotes. I wouldn't mind I18n this as well.

**Comment by meara** [comment]
@jhanggi updated :)

**Pull request merged**